ffi/src/lib.rs

```diff
@@ -388,14 +517,56 @@ fn create_client_internal(
             let redis_error = err.into();
             errors::error_message(&redis_error)
         })?;
+
+    // Wrap the runtime in an Arc so it can be shared
+    let runtime = Arc::new(runtime);
+
+    let is_subscriber = request.pubsub_subscriptions.is_some() && pubsub_callback as usize != 0;
+    let (push_tx, mut push_rx) = tokio::sync::mpsc::unbounded_channel();
+    let tx = match is_subscriber {
+        true => Some(push_tx),
+        false => None,
+    };
+
     let client = runtime
-        .block_on(GlideClient::new(ConnectionRequest::from(request), None))
+        .block_on(GlideClient::new(ConnectionRequest::from(request), tx))
         .map_err(|err| err.to_string())?;
+
+    // Create the client adapter that will be returned and used as conn_ptr
     let core = Arc::new(CommandExecutionCore {
-        client,
-        client_type,
+        client: client.clone(),
+        client_type: client_type.clone(),
     });
-    Ok(ClientAdapter { runtime, core })
+    let client_adapter = Arc::new(ClientAdapter {
+        runtime: runtime.clone(),
+        core,
+    });
+    // Clone client_adapter before moving it into the async block
+    let client_adapter_clone = Arc::into_raw(client_adapter.clone()).addr();
+
+    // If pubsub_callback is provided (not null), spawn a task to handle push notifications
+    if is_subscriber {
+        runtime.spawn(async move {
+            loop {
+                let result = push_rx.recv().await;
+                match result {
+                    None => {
+                        return;
+                    }
+                    Some(push_msg)
+                        if push_msg.kind == redis::PushKind::Message
+                            || push_msg.kind == redis::PushKind::PMessage
+                            || push_msg.kind == redis::PushKind::SMessage =>
+                    unsafe {
+                        process_push_notification(push_msg, pubsub_callback, client_adapter_clone);
+                    },
+                    _ => {}
+                }
+            }
```

Comment:
I think you can rewrite this to be a bit easier to read:
```suggestion
            while let Some(push_msg) = push_rx.recv().await {
                 if push_msg.kind == redis::PushKind::Message
                     || push_msg.kind == redis::PushKind::PMessage
                     || push_msg.kind == redis::PushKind::SMessage {
                         unsafe {
                             process_push_notification(push_msg, pubsub_callback, client_adapter_clone);
                         }
                     }
                 }
             }
```